File: tourpro/src/Components/Tour.js
import React, {useState} from 'react';
import './Tour.css';

const Tour = ({ toursId, tourImage, countries, price, information, removeTour}) =>{
    const [readMore, setReadMore ] = useState(false);
    
    return (
        <div className='whole'>
        <article className='single-tour'>
            <img id='a1' src={tourImage} alt={countries} />
            <footer>
                <div className='tour-info'>
                    <h2>{countries}</h2>
                    
                </div>
                <p>
                    {readMore ? information : `${information.substring(0,100)}`}
                    <button onClick={() => setReadMore(!readMore)}>
                        {readMore ? 'show less' : 'read more'}
                    </button>
                </p>
                <h4 className='tour-price'>{price}</h4>
                
                <button className='delete-btn' onClick={() => removeTour(toursId)}>
                    Not Interested
                </button>
            </footer>
        </article>
        </div>
    );
};
export default Tour;

File: ToursApi/Models/Tour.cs
using System;
using System.Collections.Generic;

namespace ToursApi.Models;

public partial class Tour
{
    public int ToursId { get; set; }

    public string? Countries { get; set; }

    public int? Days { get; set; }

    public string? Price { get; set; }

    public string? Information { get; set; }

    public string? Interested { get; set; }

    public string? TourImage { get; set; }
}
